packages/runtime-core/src/apiAsyncComponent.ts

```diff
@@ -118,6 +121,17 @@ export function defineAsyncComponent<
 
     __asyncLoader: load,
 
+    __asyncHydrate(el, instance, hydrate) {
+      const doHydrate = hydrateStrategy
```

Comment:
Do we need to handle exceptions since user can pass a custom strategy? If an exception occurs in `hydrateStrategy`, fall back to `hydrate`.

Reply:
Good point. Although it should be noted that even if sync errors are handled, we can't catch all possible async errors in custom strategies, so there will still be a possibility for the hydration to never happen. In other words, error handling here can only mitigate errors to a certain extent - it's ultimately the user's responsibility to provide a working strategy.